tests/test_component.py

```diff
@@ -19,3 +22,57 @@ def test_component_methods_are_valid():
     component = ExampleComponent()
     expected_methods = ["test_method"]
     assert list(component._functions().keys()) == expected_methods
+
+
+class DirectTemplateComponent(Component):
+    text = ""
+    texts = []
+
+    def __init__(self, template_string: str, **kwargs):
+        super().__init__(**kwargs)
+        self._template = Template(template_string)
+
+    def _render_template(self, template_name: str, context_variables: dict):
+        return self._template.render(context_variables)
+
+
+def test_render_model_value(app):
+    # GIVEN
+    template = """
+    <div class="row">
+        <input type="text" meld:model="text">
+        <button class="btn right" meld:click="add">+</button>
+    </div>
+    """
+    component = DirectTemplateComponent(template)
+
+    # When
+    app.config["SERVER_NAME"] = "localhost"
+    with app.app_context():
+        component.text = "hello"
+        rendered_html = component.render("DirectTemplateComponent")
+
+    # Then
+    soup = BeautifulSoup(rendered_html, features="html.parser")
+    assert soup.find("input").attrs["value"] == "hello"
+
+
+def test_render_model_defer_value(app):
+    # GIVEN
+    template = """
+    <div class="row">
+        <input type="text" meld:model.defer="text">
+        <button class="btn right" meld:click="add">+</button>
+    </div>
+    """
+    component = DirectTemplateComponent(template)
+
+    # When
+    app.config["SERVER_NAME"] = "localhost"
+    with app.app_context():
+        component.text = "hello"
+        rendered_html = component.render("DirectTemplateComponent")
+
+    # Then
+    soup = BeautifulSoup(rendered_html, features="html.parser")
+    assert soup.find("input").attrs["value"] == "hello"
```

Comment:
This tests the rendering, the test setup is not optimal but is a first draft to test this feature in general

Reply:
We'll have some work to do with the test suite. This is a great start, we can refactor it in a new issue